### ISSUE TITLE
Multi Aspect Ratio training

### ISSUE THREAD

**Issue by yjhong89** [Tencent/HunyuanVideo-I2V#59]
Hi! 

Any plan to release multi-aspect ratio training??

Or is it already supported by extracting latent with `enable_multi_aspect_ratio=True` ??

Thanks!

**Comment by Rowing0914** [comment]
I'm stuck at this and +1 from me too.

**Comment by Changlin-Lee** [comment]
Hi, multi-aspect ratio training is enabled after setting enable_multi_aspect_ratio=True in extracting latent.

Can you detailed described the question about IterableDataset? I am not getting with it.

**Comment by Rowing0914** [comment]
> Changlin-Lee
Hi, multi-aspect ratio training is enabled after setting enable_multi_aspect_ratio=True in extracting latent.
Can you detailed described the question about IterableDataset? I am not getting with it.

In my case, Yes, I’ve already enabled enable_multi_aspect_ratio=True during latent extraction.

Regarding the IterableDataset — I was referring to an issue I encountered when loading a batch of video latents. Since each sample can have a different shape (due to varying aspect ratios), torch.stack in the default collate_fn throws a runtime error.

Do you have any suggestions on how to handle variable-shaped inputs in this case? Should I write a custom collate_fn, or is there a recommended approach when using multi-aspect ratio training?

Thanks again!

**Comment by Rowing0914** [comment]
I might've figured what my issue was. Are we supposed to use `video-micro-batch-size=1` for training...? I thought we could batch videos.

**Comment by yjhong89** [comment]
For me, I set batch size =1 (video-micro-batch-size) and it works fine.
But if want to increase batch size, padding would be necessary in collate fn.
Check FastVideo repo.
- https://github.com/hao-ai-lab/FastVideo/blob/eac79b753fd6e50d3f50d2c46390311595be9471/fastvideo/dataset/latent_datasets.py#L67
